### ISSUE TITLE
Traits implementation not found

### ISSUE THREAD

**Issue by friedger** [blockstack/stacks-blockchain#2599]
**Describe the bug**
Contract rocket-market-v3 implements the tradable-trait. However, the traits endpoint returns a 404

**Steps To Reproduce**
1. Verify that [rocket-market-v3](https://explorer.stacks.co/txid/ST33GW755MQQP6FZ58S423JJ23GBKK5ZKH3MGR55N.rocket-market-v3?chain=testnet) implements the [tradables trait](https://explorer.stacks.co/txid/ST33GW755MQQP6FZ58S423JJ23GBKK5ZKH3MGR55N.tradables-trait-v1?chain=testnet).
1. Check result of https://stacks-node-api.testnet.stacks.co/v2/traits/ST33GW755MQQP6FZ58S423JJ23GBKK5ZKH3MGR55N/rocket-market-v3/ST33GW755MQQP6FZ58S423JJ23GBKK5ZKH3MGR55N/tradables-trait-v1/tradables-trait
2. See the error message.

**Screenshots**
Contract:
![Screenshot from 2021-04-21 10-48-18](https://user-images.githubusercontent.com/1449049/115525226-2fbef980-a28f-11eb-96c7-58d60546488c.png)
![Screenshot from 2021-04-21 10-47-52](https://user-images.githubusercontent.com/1449049/115525228-30579000-a28f-11eb-9e30-5b67c9e38ddc.png)

Trait:
![Screenshot from 2021-04-21 10-48-31](https://user-images.githubusercontent.com/1449049/115525257-35b4da80-a28f-11eb-9611-a1657a7db296.png)

Result:
![Screenshot from 2021-04-21 10-49-24](https://user-images.githubusercontent.com/1449049/115525309-436a6000-a28f-11eb-9b98-eb584ae1347c.png)

**Comment by jcnelson** [comment]
Yup, I think this is a bug. Thanks for reporting @friedger!

@pavitthrap I think the problem is in `handle_get_is_trait_implemented()` in `src/net/rpc.rs`:

```rust
let response =
    match chainstate.maybe_read_only_clarity_tx(&sortdb.index_conn(), tip, |clarity_tx| {
        clarity_tx.with_clarity_db_readonly(|db| {
            let analysis = db.load_contract_analysis(&contract_identifier)?;
            if analysis.implemented_traits.contains(trait_id) {
                Some(GetIsTraitImplementedResponse {
                    is_implemented: true,
                })
            } else {
                // *******************************************************************************
                // I think the line below is the problem -- this can return None prematurely if the
                // contract was declared before the trait was.  I think you'll need to load up
                // the trait-defining contract's analysis instead, and use that.
                // *******************************************************************************
                let trait_definition = analysis.get_defined_trait(&trait_id.name)?;
                let is_implemented = analysis
                    .check_trait_compliance(trait_id, trait_definition)
                    .is_ok();
                Some(GetIsTraitImplementedResponse { is_implemented })
            }
        })
    }) {
```

**Comment by pavitthrap** [comment]
Fixed in #2602
